### PR TITLE
Introduce channel APIs

### PR DESCRIPTION
**Pull request by sharath2727**
Description: Introduce the channel APIs in the PushNotifications.idl

### CODE REVIEW

**Review by danielayala94 — APPROVED** [review]

File: dev/PushNotifications/PushNotifications.idl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.ProjectReunion
+{
+    enum ChannelStatus
+    {
+       InProgress, // The request is in progress and there is no retry operation
```
Comment (hulumane) [inline dev/PushNotifications/PushNotifications.idl:8]: indentation

File: dev/PushNotifications/PushNotifications.idl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        // Low-level information about why the operation might not
+        // have worked. Apps generally do not take action based on
+        // this value, in part because the value may not be stable.
+        HRESULT ExtendedError{ get; };
+
+        // The Status of the ChannelComplete operation
+        ChannelStatus Status { get; };
+    };
+
+    static runtimeclass PushNotificationChannelManager
```
Comment (hulumane) [inline dev/PushNotifications/PushNotifications.idl:33]: PushManager

File: dev/PushNotifications/PushNotifications.idl [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{
+    enum ChannelStatus
+    {
+       InProgress, // The request is in progress and there is no retry operation
+        InProgressRetry, // The request is in progress and is in a backoff retry state
+        CompletedSuccess, // The request completed successfully
+        CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+    };
+
+    runtimeclass PushChannelResult
```
Comment (hulumane) [inline dev/PushNotifications/PushNotifications.idl:14]: ChannelResult

File: dev/PushNotifications/PushNotifications.idl [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        HRESULT ExtendedError{ get; };
+
+        // The Status of the ChannelComplete operation
+        ChannelStatus Status { get; };
+    };
+
+    static runtimeclass PushNotificationChannelManager
+    {
+        // Request a Push Channel with an encoded RemoteId from WNS. RemoteId is an AAD identifier GUID
+        static Windows.Foundation.IAsyncOperationWithProgress<PushChannelResult, PushChannelResult> CreateChannelAsync(String remoteId);
```
Comment (hulumane) [inline dev/PushNotifications/PushNotifications.idl:36]: Guid remoteId

Comment (hulumane) [reply dev/PushNotifications/PushNotifications.idl:36]: We are going with a non-string

**Review by hulumane — APPROVED** [review]

**Review by danielayala94 — APPROVED** [review]

**Pull request merged**